src/Symfony/Component/Form/ButtonBuilder.php

```diff
@@ -159,31 +187,45 @@ public function getForm()
     }
 
     /**
-     * Unsupported method.
-     *
-     * This method should not be invoked.
-     *
-     * @param string   $eventName
-     * @param callable $listener
-     * @param int      $priority
+     * {@inheritdoc}
      *
      * @throws BadMethodCallException
      */
     public function addEventListener($eventName, $listener, $priority = 0)
     {
-        throw new BadMethodCallException('Buttons do not support event listeners.');
+        if ($this->locked) {
+            throw new BadMethodCallException('ButtonBuilder methods cannot be accessed anymore once the builder is turned into a FormConfigInterface instance.');
+        }
+
+        if (isset(self::UNSUPPORTED_FORM_EVENTS[$eventName])) {
+            throw new BadMethodCallException(sprintf('Buttons do not support the "%s" form event. Use "%s" instead.', $eventName, implode('" or "', self::UNSUPPORTED_FORM_EVENTS[$eventName])));
```

Comment:
My comment above is related to this line, where `implode('" or "', self::UNSUPPORTED_FORM_EVENTS[$eventName])` displays the supported events, it confuses me.